cmd/flags/admin.go

```diff
@@ -111,17 +127,31 @@ func (s *AdminServer) serveWithListener(l net.Listener) {
 		Handler:  recoveryHandler(s.mux),
 		ErrorLog: errorLog,
 	}
+	if s.TLSConfig.Enabled {
+		tlsCfg, err := s.TLSConfig.Config(s.logger) // This checks if the certificates are correctly provided
```

Comment:
instead of having to handle the error hear, why not handle it where we call InitFromViper ?

Reply:
I did it this the same way it's done with the other servers but it seems a better design

Reply:
The error checking can be done in the `InitFromViper()` but as the server is not instantiated at that moment the tlsCfg association can only be done here.

Reply:
Done :) !